src/libraries/System.Private.CoreLib/src/System/Collections/HashHelpers.cs

```diff
@@ -90,21 +90,24 @@ public static int ExpandPrime(int oldSize)
         }
 
 #if BIT64
+        // Returns approximate reciprocal of the divisor: ceil(2**64 / divisor)
         public static ulong GetFastModMultiplier(uint divisor)
             => ulong.MaxValue / divisor + 1;
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static uint FastMod(uint value, uint divisor, ulong multiplier)
```

Comment:
Surely this is inlined even without the attribute..

Reply:
Have you verified it? This method is going to be ~20 bytes of IL. It is in the gray where the JIT may choose to not inline it.

Reply:
You are [exactly right](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4aNLtgG4x2FQxFdcEaQEExYpAE4aAbxoNnTUk3IB2Jy8fUX/9yQmFAYAWWxeLgAKZjoAbQBdBmwoAHNcAEpvAIZfHJyOSIwGBWxJDhgGAF4GQp4WAAUU3BgolPS4ugSMjT98/zrigBNeJVMoatqixubW9tw48m7e/oHJCC5UhnwOST4xSV4ZScGZqBa2tIXSZezVu/7mGyieh/y3gqKGWFxd4pqAGJ4DChCBDKKlcowNAMEZjaAwnZ7XgHI5QV59VYMD4BJ5RH5/DE4gC+OJxAHpyXFQjAMAALMEASXEkiiNPpTJZAHkxHwNrgWOZUqkfrhRjBGVxDlxIqkMgkccQAMyBKY8BhA3AgsFRQYlMoVGF6uG8cZG9abbZ/FGHGRZTEBPJY2oWrbrADuwF4GFwkyR+1tEwAVPqoStnXq6bxUnSvT7Jrqihkoimoh6476AHyZhhK0gZBgAagY5ALIZN4wL2dz+d6OP8xA8DCjMYz4ZypId/k0Ds7naAA===) .. it is 18 bytes IL, if I remove the `+ 1` to make 15 bytes of IL, it is then inlined.